[Feature] 닉네임 생성 뷰 그리기

MuchanKim opened the pull request:
## 1️⃣ PR Type
<!-- 해당하는 유형에 "x"를 입력하세요. -->
<!-- 해당하는 타입이 없다면 생성해주세요. -->

- [x] Feature: 기능 추가
- [ ] Chore: 환경 설정
- [ ] Refactor: 코드 개선
- [ ] Style: 디자인 수정
- [ ] Bugfix: 버그 수정

## 2️⃣ 작업 내용
목데이터를 사용하여 닉네임 생셩 뷰를 만들었습니다.

## 3️⃣ 관련 이슈
- #1 

## 4️⃣ 스크린샷
![CreateNicknameView](https://github.com/user-attachments/assets/af3b1419-bbad-419d-a536-e20b3b3797f6)

## 5️⃣ 체크리스트
- [x] 작성한 코드가 정상적으로 동작하나요?
- [x] 사전에 정해진 컨벤션을 지켜서 코드를 작성했나요?
- [x] Pull Request를 보내기 전 리뷰어와 할당자를 설정했나요?
- [x] 관련 이슈 번호를 작성했나요?

## 6️⃣ 리뷰 요구사항 및 추가 설명
1. 네이밍에 대한 피드백 주시면 좋을 것 같습니다. (냉정 피드백 환영)
2. CreatNicknameView.swift - 기존 닉네임 입력 후 버튼 클릭 시 버튼 이미지가 전환되게 구현하는 것 대신, 글자가 입력되면 버튼이 활성화 되도록 구현하였습니다. 
```swift
Button(action: {
    // 버튼이 활성화된 상태에서만 눌리도록
    if isButtonEnabled {
        isButtonPressed.toggle()
    }
}) {
    // 버튼의 이미지가 활성화 여부에 따라 다르게 표시
    Image(isButtonEnabled ? "startButton_pressed" : "startButton_default")
}
// 버튼이 비활성화 상태일 때는 눌리지 않도록 설정
.disabled(!isButtonEnabled)
```

3. 폴더 구조에 대한 의문점 - NicknameCardView를 Nickname/Components에 넣는게 적절한지 아니면 추천하는 방식이 있으면 피드백 부탁드립니다.

4. NicknameCardView에서 안내문까지 뷰로 작성해서 다 세분화를 시켜놨는데 괜찮은 방식일까요? text는 추후에 닉네임 변경에서 재사용할 수 있기 때문에 따로 만들어줬습니다.

MuchanKim committed:
chore: 닉네임 생성 뷰 이미지 리소스 추가 #1

*아래 파일들을  Image.xcassets에 추가하였습니다.
profile_harvang - 캐릭터 사진
maintitle - 닉네임 생성 뷰, 메인 뷰 상단에 표시되는 텍스트 이미지
startButton_default - 기본 시작 버튼
startButton_pressed - 시작 버튼을 눌렀을 때 버튼

MuchanKim committed:
feat: 닉네임 생성 카드 뷰 구현 #1
닉네임 생성 뷰에서 가운데 카드(프레임)형태의 컴포넌트를 구현하였습니다.
세부 코드에 대해서는 주석으로 첨부하였습니다!

MuchanKim committed:
feat: 닉네임 생성 뷰 그리기 1차 완료  #1

NicknameCardView를 사용하여 첫 닉네임 생성 뷰 그리기 1차 완료했습니다.
geometryReader로 비율로 레이아웃 구성하였으나 일부 resource에는 적용이 안된 상태입니다.
12.9, 13inch에서 레이아웃은 정상적으로 배치되었기에 임시적으로 작업 마치고 다음 작업 이어가겠습니다.

MuchanKim committed:
chore: NicknameCardView 주석 수정 #1

굳이 필요 없는 주석들을 삭제했습니다.

delightPIP commented:
피드백

 ***** iPad 11 에서는 모두 깨져서 보기가 어려웠습니다. 
아이패드 상용화시에는 11버전까지도 모두 맞춰야합니다. 
이 부분에 대한 사안을 해결해야할 것 같습니다.

1. Figma 에는 harbang 으로 되어 있는데 harvang 으로 되어있음
하나로 통일 하면 좋을 듯 합니다. 저 같은 경우에는 캐릭터마다 번호(백엔드에서 부여받을)로만 표시해두었는데 나중에 상의해서 통일시키면 될 것 같네요.

리소스 관리 시에 했으면 하는건 공통을 뽑아서 앞에 배치했으면 합니다.

> 버튼 이미지일 경우 -> button_XXX, btn_XXX
> 버튼 이미지(활성화/비활성화) -> button_XXX_default, btn_XXX_pressed
> 캐릭터 이미지일 경우 -> character_XXX
> 게임의 버튼 이미지일 경우 -> button_game_XXX, btn_game_XXX

이런 식으로 구성하면 어떨까 싶어요

2. Input 할때 UI/UX적으로 좋은 버튼이네요, 키보드 올라올 때의 이슈만 해결한다면 닉네임 수정 시에도 공통 사용 가능할 것 같습니다. 테스트 시에 텍스트가 5글자 이상 들어가도 제한이 되지 않고 있었습니다. 이 부분은 valid 를 위해 제한해주는 게 좋을 것 같아요.
3. 해당 기능에서만 사용한다면 Component 를 내부에 넣으면 될 것 같고, 다른 부분에서도 쓴다면 상위 Helper 디렉토리에 추가되는게 맞을 것 같습니다.
4. Component 라는 뜻 자체가 재사용이 가능한 반복요소를 넣는 걸로 알고 있습니다. 해당 카드뷰는 추후 닉네임 수정 시에도 쓸 수 있어서 Component 에 넣어도 될 것 같은데 또 성격자체는 Views 안에서도 맞을 것 같네요. 이 부분은 검토하여 선택하시면 좋을 것 같습니다.

MuchanKim committed:
chore: 이미지 리소스 명 변경 및 이에 따른 일부 코드 수정 #1

요구사항 대로 공통을 앞에 두고 네이밍을 하였습니다.
캐릭터의 경우 사용도가 다양해서
<character><사용도><캐릭터이름> 양식으로 사용했고
예시는 character_card_harbang, character_splash_rabong입니다.
harvang -> harbang 수정했습니다. figma 이름대로 통일하면 될 것 같습니다.
text를 붙이는 경우에는 <text><사용도>로 만들었습니다.
예시는 text_maintitle, text_splash 입니다.

MuchanKim committed:
feat: textfield 5글자 제한 기능 #1

닉네임 입력칸에 5글자 이상 입력 못하도록 구현하였습니다.
추가적으로 onChange 메서드가 iOS 17이후 버전에만 지원 해줘서 버전 별로 조건부 컴파일을 하도록 지정해주었습니다.

MuchanKim committed:
feat: 반응형 레이아웃 구현 #1

ipad mini, ipad 11inch 모델에서 레이아웃이 깨지는 문제가 있었는데, 수정하였습니다.
자세한 이미지는 이슈에 이미지로 첨부하겠습니다.

MuchanKim commented:
- 네이밍 컨벤션에 따라 수정했습니다.
- 닉네임 입력창 5자 이상 입력 안되도록 수정하였습니다.
- GeometryReader사용해서 비율로 레이아웃 잡았습니다. 코드 실행시켜 보셔도 되지만 아래 이미지 첨부했습니다. 

검토 부탁드립니다.
### 예시 
```swift
HStack(spacing: geometry.size.width * 0.05) {
    Image(selectedCharacter)
        .resizable()
        .scaledToFit()
        .frame(width: geometry.size.width * 0.415)

    VStack(alignment: .center, spacing: geometry.size.height * 0.03)
    // 생략
    }
}
```

![image](https://github.com/user-attachments/assets/a707d587-40ce-45c0-96bc-79471bf468b7)

MuchanKim committed:
feat: 닉네임 생성 뷰 그리기 완료 #1

DeviceScaleView 참고해서 12.9인치 맞췄고, 각 기기에서 레이아웃 맞춰지는거 확인했습니다.

MuchanKim commented:
<img src="https://github.com/user-attachments/assets/9e9b7fdc-553c-4a5a-9615-fd3d6753de3e" width="550" />

레이아웃 맞췄습니다!! (최종)
